packages/oft-evm/contracts/NativeOFTAdapter.sol

```diff
@@ -0,0 +1,114 @@
+     */
+    function _debit(
+        address _from,
+        uint256 _amountLD,
+        uint256 _minAmountLD,
+        uint32 _dstEid
+    ) internal virtual override returns (uint256 amountSentLD, uint256 amountReceivedLD) {
+        // @dev Lock native funds by moving them into this contract from the caller.
+        // No need to transfer here since msg.value moves funds to the contract.
+        (amountSentLD, amountReceivedLD) = _debitView(_amountLD, _minAmountLD, _dstEid);
+    }
+
+    /**
+     * @dev Credits native to the specified address.
+     * @param _to The address to credit the native to.
+     * @param _amountLD The amount of native to credit.
+     * @dev _srcEid The source chain ID.
+     * @return amountReceivedLD The amount of native ACTUALLY received.
+     */
+    function _credit(
+        address _to,
+        uint256 _amountLD,
+        uint32 /*_srcEid*/
+    ) internal virtual override returns (uint256 amountReceivedLD) {
+        // @dev Unlock the tokens and transfer to the recipient.
+        (bool success, ) = payable(_to).call{value: _amountLD}("");
+        if (!success) {
+            revert CreditFailed();
+        }
+
+        // @dev In the case of NON-default NativeOFTAdapter, the amountLD MIGHT not be == amountReceivedLD.
+        return _amountLD;
+    }
+
+    function send(
+        SendParam calldata _sendParam,
+        MessagingFee calldata _fee,
+        address _refundAddress
+    ) external payable override returns (MessagingReceipt memory msgReceipt, OFTReceipt memory oftReceipt) {
+        if (msg.value < _sendParam.amountLD) {
+            revert InsufficientMessageValue(msg.value, _sendParam.amountLD);
+        }
+        
+        MessagingFee memory feeWithExtraAmount = MessagingFee({
+            nativeFee: _fee.nativeFee,
+            lzTokenFee: _fee.lzTokenFee
+        });
+
+        uint256 remainingMsgValue = msg.value - _sendParam.amountLD;
+        feeWithExtraAmount.nativeFee = remainingMsgValue;
+
+        (uint256 amountSentLD, uint256 amountReceivedLD) = _debit(
+            msg.sender,
+            _sendParam.amountLD,
+            _sendParam.minAmountLD,
+            _sendParam.dstEid
+        );
+
+        // @dev Builds the options and OFT message to quote in the endpoint.
+        (bytes memory message, bytes memory options) = _buildMsgAndOptions(_sendParam, amountReceivedLD);
+
+        // @dev Sends the message to the LayerZero endpoint and returns the LayerZero msg receipt.
+        msgReceipt = _lzSend(_sendParam.dstEid, message, options, feeWithExtraAmount, _refundAddress);
+        
+        // @dev Formulate the OFT receipt.
+        oftReceipt = OFTReceipt(amountSentLD, amountReceivedLD);
+
+        emit OFTSent(msgReceipt.guid, _sendParam.dstEid, msg.sender, amountSentLD, amountReceivedLD);
+    }
+
+    function _payNative(uint256 _nativeFee) internal override returns (uint256 nativeFee) {
```

Comment:
override this to just do nothing

Reply:
I would also comment the reason for muting this out was that the assertion is now done on the send higher up in this contract